azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -1,631 +1,657 @@
+
+
+RequestParams = namedtuple("RequestParams", ["json_payload", "request_headers", "timeout"])
+
+
+class _KustoClientBase:
+    _mgmt_default_timeout = timedelta(hours=1, seconds=30)
+    _query_default_timeout = timedelta(minutes=4, seconds=30)
+    _streaming_ingest_default_timeout = timedelta(minutes=10)
+
+    def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str]):
+        self._kcsb = kcsb
+        if not isinstance(kcsb, KustoConnectionStringBuilder):
+            self._kcsb = KustoConnectionStringBuilder(kcsb)
+        self._kusto_cluster = self._kcsb.data_source
+
+        # Create a session object for connection pooling
+        self._mgmt_endpoint = "{0}/v1/rest/mgmt".format(self._kusto_cluster)
+        self._query_endpoint = "{0}/v2/rest/query".format(self._kusto_cluster)
+        self._request_headers = {"Accept": "application/json", "Accept-Encoding": "gzip,deflate", "x-ms-client-version": "Kusto.Python.Client:" + VERSION}
+
+    @staticmethod
+    def _kusto_parse_by_endpoint(endpoint: str, response_json: List[dict]) -> KustoResponseDataSet:
+        if endpoint.endswith("v2/rest/query"):
+            return KustoResponseDataSetV2(response_json)
+        return KustoResponseDataSetV1(response_json)
+
+    def _build_execute_request_params(
+        self, database: str, payload: io.IOBase, properties: ClientRequestProperties, query: str, timeout: timedelta
+    ) -> RequestParams:
+        request_headers = copy(self._request_headers)
+        json_payload = None
+        if not payload:
+            json_payload = {"db": database, "csl": query}
+            if properties:
+                json_payload["properties"] = properties.to_json()
+
+            client_request_id_prefix = "KPC.execute;"
+            request_headers["Content-Type"] = "application/json; charset=utf-8"
+        else:
+            if properties:
+                request_headers.update(json.loads(properties.to_json())["Options"])
+
+            client_request_id_prefix = "KPC.execute_streaming_ingest;"
+            request_headers["Content-Encoding"] = "gzip"
+        request_headers["x-ms-client-request-id"] = client_request_id_prefix + str(uuid.uuid4())
+        if properties is not None:
+            if properties.client_request_id is not None:
+                request_headers["x-ms-client-request-id"] = properties.client_request_id
+            if properties.application is not None:
+                request_headers["x-ms-app"] = properties.application
+            if properties.user is not None:
+                request_headers["x-ms-user"] = properties.user
+        if properties:
+            timeout = properties.get_option(ClientRequestProperties.request_timeout_option_name, timeout)
+
+        return RequestParams(json_payload, request_headers, timeout)
+
+    @staticmethod
+    def _build_kusto_service_error(payload: io.IOBase, response: _get_dynamic_response_type(), response_json: List[dict]) -> KustoServiceError:
```

Comment:
this `_get_dynamic_response_type` beats the purpose of annotations. annotations are here to help me read the code and make sure I don't make typing mistakes. if they are resolved in runtime it does not help at all.
please revert this.

if you want a dynamic type, just use `Any` and duck type it

Reply:
I prefered to duplicate this specific method, so the typing will be more accurate and informative